src/use-background-queue/use-background-queue.ts

```diff
@@ -0,0 +1,81 @@
+import { Controller } from '@hotwired/stimulus'
+
+class BackgroundQueueController extends Controller {
+  static queue: string[] | BackgroundQueueOptions[] = []
+}
+
+export interface BackgroundQueueOptions {
+  maxWait?: number,
+  name?: string
+}
+
+const defaultOptions = {}
+const idleCallbackIds: number[] = []
+
+const ric = (fn: Function, maxWait?: number) => {
+  // https://developer.chrome.com/blog/using-requestidlecallback/
+  if (!window.requestIdleCallback) {
+    const start = Date.now()
+
+    return function (this: any) {
+      const args = arguments
+      const context = this
+
+      const timeoutId = window.setTimeout(() => {
+        fn.call(context, ...args, { didTimeout: false, timeRemaining: () => Math.max(0, 50 - (Date.now() - start)) })
+      }, 1)
+
+      idleCallbackIds.push(timeoutId)
+    }
+  }
+
+  return function (this: any) {
+    const args = arguments
+    const context = this
+
+    const idleCallbackId = window.requestIdleCallback(deadline => fn.call(context, ...args, deadline), { timeout: maxWait })
+
+    idleCallbackIds.push(idleCallbackId)
+  }
+}
+
+const cancelRic = (id: number) =>
```

Comment:
The same for the `cancelIdleCallback`. Also, I would check for `window.cancelIdleCallback` instead of `window.requestIdleCallback` here